source/common/buffer/buffer_impl.cc

```diff
@@ -94,7 +94,33 @@ void OwnedImpl::move(Instance& rhs, uint64_t length) {
 }
 
 int OwnedImpl::read(int fd, uint64_t max_length) {
-  return evbuffer_read(buffer_.get(), fd, max_length);
+  constexpr uint64_t MaxSlices = 2;
+  RawSlice slices[MaxSlices];
+  uint64_t num_slices = reserve(max_length, slices, MaxSlices);
+  struct iovec iov[num_slices];
+  int num_iov = 0;
+  for (uint64_t i = 0; i < num_slices && max_length != 0; i++) {
+    num_iov++;
```

Comment:
Doesn't this always have the same value as i at the end of the loop? Can we remove i and just use this to simplify?